### PR TITLE
chore: update to new multiformats

### PR DESCRIPTION
**Pull request by achingbrain**
Uses new CID class.  Also swaps out various chai deps for version
bundled with aegir, updates aegir and fixes tests that start services
without tearing them down which stops the tests from finishing.

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -204,9 +204,13 @@ exports.withTimeout = (asyncFn, time) => {
     return Promise.race([
       asyncFn(...args),
       new Promise((resolve, reject) => {
-        setTimeout(() => {
+        const timeout = setTimeout(() => {
           reject(errcode(new Error('Async function did not complete before timeout'), 'ETIMEDOUT'))
         }, time)
+
+        if (timeout && timeout.unref) {
```
Comment (vasco-santos) [inline src/utils.js:211]: `setTimeout` return a number, `timeoutId`. Type checker is failing here as a consequence

File: src/content-routing/index.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const Query = require('../query')
 const utils = require('../utils')
 
 /**
- * @typedef {import('cids')} CID
+ * @typedef {import('multiformats').CID} CID
```
Comment (vasco-santos) [inline src/content-routing/index.js:13]:
```suggestion
 * @typedef {import('multiformats/cid').CID} CID
```

File: src/message/dht.d.ts [hunk 3]
```diff
@@ -28,19 +28,34 @@ export class Record implements IRecord {
     constructor(p?: IRecord);
 
     /** Record key. */
-    public key: Uint8Array;
+    public key?: (Uint8Array|null);
```
Comment (vasco-santos) [inline src/message/dht.d.ts:31]: was this generated?

File: src/providers.js [hunk 4]
```diff
@@ -10,7 +10,7 @@ const c = require('./constants')
 const utils = require('./utils')
 
 /**
- * @typedef {import('cids')} CID
+ * @typedef {import('multiformats').CID} CID
```
Comment (vasco-santos) [inline src/providers.js:13]:
```suggestion
 * @typedef {import('multiformats/cid').CID} CID
```

File: src/rpc/handlers/add-provider.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-const CID = require('cids')
+const { CID } = require('multiformats')
```
Comment (vasco-santos) [inline src/rpc/handlers/add-provider.js:3]:
```suggestion
const { CID } = require('multiformats/cid')
```

File: src/rpc/handlers/get-providers.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-const CID = require('cids')
+const { CID } = require('multiformats')
```
Comment (vasco-santos) [inline src/rpc/handlers/get-providers.js:3]:
```suggestion
const { CID } = require('multiformats/cid')
```

File: src/index.js [hunk 7]
```diff
@@ -254,12 +258,12 @@ class KadDHT extends EventEmitter {
    *
    * @returns {Promise<void>}
    */
-  stop () {
+  async stop () {
     this._running = false
     this.randomWalk.stop()
-    this.providers.stop()
+    await this.network.stop()
```
Comment (vasco-santos) [inline src/index.js:264]: This does not need to be a promise, but `network.stop` and `network.start` need to be fixed. Registrar does not return a promise anymore: https://github.com/libp2p/js-libp2p/blob/master/src/registrar.js#L87

File: test/providers.spec.js [hunk 8]
```diff
@@ -1,15 +1,13 @@
 /* eslint-env mocha */
 'use strict'
 
-const chai = require('chai')
-chai.use(require('dirty-chai'))
-const expect = chai.expect
+const { expect } = require('aegir/utils/chai')
 const { MemoryDatastore } = require('interface-datastore')
-const CID = require('cids')
+const { CID } = require('multiformats')
```
Comment (vasco-santos) [inline test/providers.spec.js:6]:
```suggestion
const { CID } = require('multiformats/cid')
```

File: test/utils/create-values.js [hunk 9]
```diff
@@ -1,16 +1,16 @@
 'use strict'
 
-const multihashing = require('multihashing-async')
-const CID = require('cids')
+const { CID } = require('multiformats')
```
Comment (vasco-santos) [inline test/utils/create-values.js:3]:
```suggestion
const { CID } = require('multiformats/cid')
```

**Review by vasco-santos — CHANGES_REQUESTED** [review]
Mostly minor stuff, some are learnings from @rvagg cid import flags in ipfs-repo

File: src/utils.js (hunk 1 above)
Comment (achingbrain) [reply src/utils.js:211]: In node it returns a [Timeout](https://nodejs.org/api/timers.html#timers_class_timeout) which has an `unref` method property.  Weird, maybe we need to tell ts that it's used under node.

File: src/message/dht.d.ts (hunk 3 above)
Comment (achingbrain) [reply src/message/dht.d.ts:31]: Yes, I'm not sure why this changed, new patch release of protobufjs I guess?

File: src/index.js (hunk 7 above)
Comment (achingbrain) [reply src/index.js]: I've removed all of the unnecessary async.

File: src/utils.js (hunk 1 above)
Comment (achingbrain) [reply src/utils.js]: I'm managed to refactor this away after removing the unnecessary async and using the `p-timeout` module instead of `Promise.race`.

**Comment by achingbrain** [comment]
I've marked this as a draft so we can merge it when #3556 is green as it'll bubble everything all the way up.  Please do continue review and/or approve as appropriate.

File: src/index.js [hunk 10]
```diff
@@ -29,7 +29,7 @@ const Record = libp2pRecord.Record
  * @typedef {import('interface-datastore').Datastore} Datastore
  * @typedef {import('libp2p/src/dialer')} Dialer
  * @typedef {import('libp2p/src/registrar')} Registrar
- * @typedef {import('cids')} CID
+ * @typedef {import('multiformats').CID} CID
```
Comment (vasco-santos) [inline src/index.js:32]:
```suggestion
 * @typedef {import('multiformats/cid').CID} CID
```

**Review by vasco-santos — CHANGES_REQUESTED** [review]

**Pull request merged**